pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,168 @@
+
+## Step 1: Setup OIDC provider in AWS account
+
+First we'll set up an AWS IAM OpenID Connect (OIDC) provider using the following configuration.
+
+  URL: `https://agent.buildkite.com`
+
+  Audience: `sts.amazonaws.com`
+
+## Step 2: Create a new (or update an existing) IAM role to use with your pipeline(s)
+
+In this example the IAM role is named: `compute-ssm-oidc-example`. When creating the role you must `Select Custom Trust Policy`.
+
+Set the Principal to be federated via the IAM OIDC provider we just created, and add some conditions against using this IAM role for additional security using the format of the OIDC token subject provided by Buildkite.
+
+This allows us to limit use of this IAM role to named Organizations, Pipelines, and optionally branches, commits and steps. Format is
+ `organization:ORGANIZATION_SLUG:pipeline:PIPELINE_SLUG:ref:REF:commit:BUILD_COMMIT:step:STEP_KEY`
+
+You can use a wildcard `*` to replace any subjects you don’t want to set limits on.
+
+You can also add multiple token subjects as a list if you want to use the same IAM role for multiple pipelines, or if you want a variety of branches, commits and steps enabled.
+
+If you have a public IP address associated with your Buildkite Agent's, you can also set a condition on the source IP address and further restrict access to your IAM role.
+
+Example Trust Policy:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "agent.buildkite.com:aud": "sts.amazonaws.com"
+                },
+                "StringLike": {
+                    "agent.buildkite.com:sub": "organization\:ORGANIZATION_SLUG\:pipeline:PIPELINE_SLUG\:ref\:REF\:commit\:BUILD_COMMIT\:step\:STEP_KEY"
+                },
+                "IpAddress": {
+                    "aws:SourceIp": [
+                        "AGENT_PUBLIC_IP_ONE",
+                        "AGENT_PUBLIC_IP_TWO"
+                    ]
+                }
+
+            }
+        }
+    ]
+}
+```
+
+Update the following sections:
+
+- In the Principal, replace `AWS_ACCOUNT_ID`  with your account ID.
+- In the Condition on the subject, replace:
+
+    * `ORGANIZATION_SLUG` with your Buildkite Organization
+    * `PIPELINE_SLUG` with your Pipeline
+    * `REF` - this is commonly replaced with `refs/heads/main` to enforce only the main branch using the IAM role  or `refs/tags/*` for only tagged releases able to deploy, or a wildcard if we want all branches to use it.
+    * `BUILD_COMMIT` - this is commonly replaced with a wildcard `*`
+    * `STEP_KEY` - this is commonly replaced with a wildcard `*`
```

Comment:
There might be a nicer way to format "replace these parts of the code snippet above" than what I've put in here, but I leave that to your expert opinion!

Reply:
Great thanks for this :-) I've reworked this somewhat—you'll see in my next commit, but it'd be ideal if you could answer my remaining queries below.